includes/options/class-amp-options-menu.php

```diff
@@ -71,6 +97,142 @@ public function add_menu_items() {
 		}
 	}
 
+	/**
+	 * Render theme support.
+	 *
+	 * @since 1.0
+	 */
+	public function render_theme_support() {
+		$theme_support = AMP_Options_Manager::get_option( 'theme_support' );
+
+		$support_args = get_theme_support( 'amp' );
+
+		$theme_support_mutable = (
+			empty( $support_args )
+			||
+			! empty( $support_args[0]['__added_via_option'] )
+		);
+		if ( ! $theme_support_mutable ) {
+			if ( amp_is_canonical() ) {
+				$theme_support = 'native';
+			} else {
+				$theme_support = 'paired';
+			}
+		}
+
+		$should_have_theme_support = in_array( get_template(), array( 'twentyfifteen', 'twentysixteen', 'twentyseventeen' ), true );
+		?>
+		<fieldset>
+			<?php if ( current_theme_supports( 'amp' ) && ! $theme_support_mutable ) : ?>
+				<div class="notice notice-info notice-alt inline">
+					<p><?php esc_html_e( 'Your active theme has built-in AMP support.', 'amp' ); ?></p>
+				</div>
+			<?php elseif ( $should_have_theme_support ) : ?>
+				<div class="notice notice-success notice-alt inline">
+					<p><?php esc_html_e( 'Your active theme is known to work well in paired or native mode.', 'amp' ); ?></p>
+				</div>
+			<?php endif; ?>
+			<dl>
+				<dt>
+					<input type="radio" id="theme_support_disabled" name="<?php echo esc_attr( AMP_Options_Manager::OPTION_NAME . '[theme_support]' ); ?>" value="disabled" <?php checked( $theme_support, 'disabled' ); ?> <?php disabled( ! $theme_support_mutable ); ?>>
+					<label for="theme_support_disabled">
+						<strong><?php esc_html_e( 'Disabled', 'amp' ); ?></strong>
```

Comment:
If somebody looked at this really quickly, it might look like AMP is disabled, not just theme support:
<img width="1432" alt="classic-templates" src="https://user-images.githubusercontent.com/4063887/41018126-ecc06fc2-691d-11e8-8245-3596c3d91fde.png">

Maybe "Disabled" could be "Classic Templates (Disabled)."

Reply:
That's a great point. Agreed.

And “Paired” doesn't really make sense to users at first look either, nor does “Native”.

Maybe:

* Classic Templates (Legacy)
* Active Theme's Templates with Separate URLs (Paired)
* Active Theme's Templates without Separate URLs (Canonical/Native)

I don't feel great about these these latter two.

Reply:
Renamed Disabled to Classic in 5bae33e. Needs more iteration.

Reply:
To look at it from a different angle, perhaps we should consider using an other term than "Theme support". I understand the reasoning behind it as it is defined using `add_theme_support` but what this feature really does is controlling the AMP Mode. So perhaps the Setting itself should be called "AMP Mode" and then have:
- Legacy
- Paired
- Native

Reply:
Yes, that makes perfect sense. 👍 

Maybe “Template Mode” would be better since it's clearly related to AMP already.